03 Writing Algorithms/03 Securities/99 Asset Classes/09 Index Options/01 Requesting Data/01 Universes/12 Examples.php

```diff
@@ -1,9 +1,138 @@
+<p>The following examples demonstrate some common practices for requesting Index Options universe data.</p>
+
 <h4>Example 1: Selecting for 0DTE Contracts</h4>
 
-<p><span class='new-term'>0DTE Options</span> are Option contracts that expire on the same day you trade them. To create a universe with 0DTE Options, call the <code class="csharp">SetFilter</code><code class="python">set_filter</code> method with the following argument:</p>
+<p><span class='new-term'>0DTE Options</span> are option contracts that expire on the same day they are traded. To create a universe with 0DTE options, call the <code class="csharp">SetFilter</code> or <code class="python">set_filter</code> method. In this example, we will select 0DTE contracts within 1 strike price level and a Delta between 0.25 and 0.75. Then, we will go long on the highest strike price call contract of the SPX index and hold until expiry.</p> 
 
 <div class="section-example-container">
-    <pre class="csharp">option.SetFilter(u => u.IncludeWeeklys().Expiration(0, 0).Strikes(minStrike, maxStrike));</pre>
-    <pre class="python">option.set_filter(lambda u: u.include_weeklys().expiration(0, 0).strikes(min_strike, max_strike))
-</pre>
+    <pre class="csharp">private Option _option;
+
+public override void Initialize()
+{
+    SetStartDate(2023, 1, 1);
+    SetEndDate(2024, 1, 1);
+    SetCash(100000);
+
+    // Subscribe to the option chain.
+    _option = AddIndexOption("SPX", Resolution.Daily);
+
+    // Filter the option universe to only select 0DTE options.
+    _option.SetFilter(u => u.IncludeWeeklys().Expiration(0, 0).Strikes(-1, 1));
+
+    // Filter the option universe by Delta.
+    _option.SetFilter(optionFilterUniverse => optionFilterUniverse.Delta(0.25m, 0.75m));
+}
+
+public override void OnData(Slice slice)
+{
+    if (Portfolio.Invested)
+    {
+        return;
+    }
+
+    // Get the option chain data.
+    if (!slice.OptionChains.TryGetValue(_option.Symbol, out var chain))
+    {
+        return;
+    }
+
+    // Select the call Option contract.
+    var calls = chain.Where(contract => contract.Right == OptionRight.Call).ToList();
+    if (calls.Count < 1) return;
+
+    // Sorted the contracts according to their strike prices.
+    calls = calls.OrderBy(x => x.Strike).ToList();
```

Comment:
We can condense this logic to:
```csharp
var call = chain.Where(contract => contract.Right == OptionRight.Call).OrderBy(x => x.Strike).FirstOrDefault();
if (call != null) Buy(call.Symbol, 1);
```
The Python version is good, but we should leverage C# features.